src/dosext/mfs/lfn.c

```diff
@@ -1133,18 +1134,51 @@ static int mfs_lfn_(void)
 		call_dos_helper(0x6c);
 		break;
 	}
-	case 0xa0: /* get volume info */
-		drive = build_posix_path(fpath, src, 0);
-		if (drive < 0)
-			return drive + 2;
+	case 0xa0: { /* get volume info */
+
+		struct get_volume_info *s;
+
+		/*
+		 * input:
+		     AX = 71A0h
+		     CX = size of buffer
+                     DS:DX -> ASCIZ root name (e.g. "C:\")
+                     ES:DI -> buffer for file system name
+		   return:
+		     AX
+		     BX
+		     CX
+		     DX
+		     buffer filled
+
+	         */
+
+		src = MK_FP32(_DS, _DX);
+		dest = SEGOFF2LINEAR(_ES, _DI);
 		size = _CX;
+
+		s = lowmem_alloc(sizeof(*s));
+		_CX = sizeof(*s);
+		_DX = 0x0100 | (uint8_t)(toupper(src[0]) - 'A');
+		_ES = DOSEMU_LMHEAP_SEG;
+		_DI = DOSEMU_LMHEAP_OFFS_OF(s);
+
+		// do int2f/11a0
+		_AX = 0x11a0;
```

Comment:
So I want to call an int2f function from here. Do I need to create something like `call_dos_helper()`, or is there a better way of doing this?

Reply:
You can use `do_int_call_back(0x2f);`,
or if you can't - it will assert.

Reply:
Cool, will try that out, thanks!